### ISSUE TITLE
F-Droid can't build

### ISSUE THREAD

**Issue by licaon-kter** [Helium314/SCEE#449]
ref: https://gitlab.com/fdroid/fdroiddata/-/jobs/4990713116

Maybe these 2 pieces of info help? https://f-droid.org/docs/Reproducible_Builds/#png-crushcrunch

/LE: fyi https://gitlab.com/fdroid/fdroiddata/-/commit/2665471d4482c8cb6a137b5567730bcfda9683b4

**Comment by Helium314** [comment]
With `cruncherEnabled` SCEE doesn't build:

```
Script compilation error:

  Line 75:         cruncherEnabled = false
                   ^ Unresolved reference: cruncherEnabled
```

However, when going back to the old Java version included in older versions of Android Studio, I get reproducible differences in the same files (can't compare because I'm not allowed to open the link to the artifact).
So I switched SCEE to use the old version and updated the APK.

I want to avoid disabling creation of PNG files from vector drawables completely. The files are automatically created because of performance issues with certain complicated vector drawables, and not because of compatibility (API < 21 not supported anyway).
See e.g. https://github.com/Helium314/SCEE/blob/modified/app/src/main/res/drawable/ic_flag_ec.xml (source of `qcP.png`)

**Comment by mnalis** [comment]
so, any resolution on this? It is still stuck in f-droid on `53.32` :disappointed:  (https://gitlab.com/fdroid/fdroiddata/-/blob/master/metadata/de.westnordost.streetcomplete.expert.yml?ref_type=heads#L103)

**Comment by Helium314** [comment]
You can still upgrade using newer versions from GitHub, so it's not that much of a problem.

Anyway, SC(EE) now switched to gradle 8, which requires the use of Java 17 (which I had used for the non-reproducible build). So the issues with different PNGs might be gone now.

btw with
```
    defaultConfig {
        vectorDrawables.generatedDensities = []
    }
```
SCEE doesn't build, so with both PNG workarounds not working I can only hope it's reproducible now.

**Comment by licaon-kter** [comment]
@Helium314 so 54.11 needs Java17? Will try locally...

**Comment by licaon-kter** [comment]
```
DOES NOT VERIFY
ERROR: APK Signature Scheme v2 signer #1: APK integrity check failed. CHUNKED_SHA256 digest mismatch. Expected: <6fe6b461562f9b8b6acc52ef48d65ae08500eb083957b18fb5556cc048ea8d44>, actual: <c0a3fd348408adf2f5dbc607bfd2439c77a250b546b860f59ded565a580314f4>
2023-09-27 08:03:02,079 ERROR: 
/tmp/tmp6cl3jiut/sigcp_de.westnordost.streetcomplete.expert_5403.apk: 
2023-09-27 08:03:02,079 INFO: ...NOT verified - /tmp/tmp6cl3jiut/sigcp_de.westnordost.streetcomplete.expert_5403.apk
2023-09-27 08:03:05,984 DEBUG: > diff -r /tmp/tmp6cl3jiut/unsigned_binaries_de.westnordost.streetcomplete.expert_5403.binary /tmp/tmp6cl3jiut/_tmp_tmp6cl3jiut_sigcp_de.westnordost.streetcomplete.expert_5403
2023-09-27 08:03:06,563 DEBUG: removing unsigned/de.westnordost.streetcomplete.expert_5403.apk
2023-09-27 08:03:06,578 DEBUG: removing unsigned/binaries/de.westnordost.streetcomplete.expert_5403.binary.apk
2023-09-27 08:03:06,839 ERROR: Could not build app de.westnordost.streetcomplete.expert: compared built binary to supplied reference binary but failed
==== detail begin ====
verification of APK with copied signature failed
Comparing reference APK to APK with copied signature...
Unexpected diff output:
Binary files /tmp/tmp6cl3jiut/unsigned_binaries_de.westnordost.streetcomplete.expert_5403.binary/content/res/-2E.png and /tmp/tmp6cl3jiut/_tmp_tmp6cl3jiut_sigcp_de.westnordost.streetcomplete.expert_5403/content/res/-2E.png differ
Binary files /tmp/tmp6cl3jiut/unsigned_binaries_de.westnordost.streetcomplete.expert_5403.binary/content/res/0wA.png and /tmp/tmp6cl3jiut/_tmp_tmp6cl3jiut_sigcp_de.westnordost.streetcomplete.expert_5403/content/res/0wA.png differ
Binary files /tmp/tmp6cl3jiut/unsigned_binaries_de.westnordost.streetcomplete.expert_5403.binary/content/res/1_r.png and /tmp/tmp6cl3jiut/_tmp_tmp6cl3jiut_sigcp_de.westnordost.streetcomplete.expert_5403/content/res/1_r.png differ
...
more PNGs
````

**Comment by Helium314** [comment]
Alright, then you can disable 54.1(1) right away.
I won't have time to work on this for the next 2-3 weeks.

**Comment by licaon-kter** [comment]
I've disabled autoupdates until ready: https://gitlab.com/fdroid/fdroiddata/-/commit/d673cf01b39b7e1ec4e5af540632bc77fcddb38d

Do ping us to check as needed

**Comment by Helium314** [comment]
Hmm ok, one last try: I replaced the APK by one built by GitHub actions. I get differences in a bunch of PNG images compared to my previous build, so I hope it works compared to your build.
(plus if it you can reproduce this, I have an easier way of finding/fixing the cause)

**Comment by licaon-kter** [comment]
I've rebuilt e985190b4a877c8e78c133b5ce7917e1d5b80199 and now it was reproducible, it means you've found the source of the non-determinism?

**Comment by mnalis** [comment]
> I've rebuilt https://github.com/Helium314/SCEE/commit/e985190b4a877c8e78c133b5ce7917e1d5b80199 and now it was reproducible, it means you've found the source of the non-determinism?

So SCEE 54.11 builds reproducibly now on f-droid @licaon-kter, if I understand you correctly? 

Does it mean `AutoUpdateMode` can be restored back to `Version` from `None`? (or does it need something else to be done before that can happen?) 

_I'm confused as https://gitlab.com/fdroid/fdroiddata/-/commits/d5f165f7da1b6e50e694aedfd3498203dd2d7f6a/metadata/de.westnordost.streetcomplete.expert.yml adds explicit disable for `54.11` linking to this issue, yet https://github.com/Helium314/SCEE/commit/e985190b4a877c8e78c133b5ce7917e1d5b80199 (which you say builds reproducibly now for you) _has_ tag `54.11` in SCEE repo?_

**Comment by licaon-kter** [comment]
@mnalis next version might be, not yet ready

**Comment by Helium314** [comment]
I wasn't able to fix the issues with the PNG files, but I'll simply use APKs built by GitHub for releases.
Just released 54.12, now the builds should be reproducible again.

**Comment by licaon-kter** [comment]
Thanks https://gitlab.com/fdroid/fdroiddata/-/commit/8d10f3f4b36d32136b71b3bcdec47b48711a87a2
